### ISSUE TITLE
Destructuring assignment default values

### ISSUE THREAD

**Issue by kentcdodds** [6to5/6to5#397]
I think this is part of the spec (though I couldn't seem to find it) based on [this note](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#See_also) comment on MDN (which points to [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=932080)).

Here's an example use case:

```js
function foo({name = 'Mary', age: 25}) {
  console.log(name, age);
}

foo({
  name: 'Joe',
  hairColor: 'Blonde'
}); // logs "Joe 25"

foo({
  name: 'Lucy',
  age: 16
}); // logs "Lucy 16"
```

Does anyone know if this is going to actually be implemented in the spec? If so, will this make it into 6to5?

**Comment by sebmck** [comment]
See 6to5/6to5#230, currently waiting on upstream acorn to support it marijnh/acorn#181.

**Comment by kentcdodds** [comment]
:+1:

**Comment by fkling** [comment]
FYI, [this is where it's defined in the spec](https://people.mozilla.org/~jorendorff/es6-draft.html#sec-destructuring-binding-patterns).


>SingleNameBinding [Yield, GeneratorParameter] : 
> [+GeneratorParameter] BindingIdentifier[Yield] Initializer[In]opt
> [~GeneratorParameter] BindingIdentifier[?Yield] Initializer[In, ?Yield]opt

**Comment by sebmck** [comment]
Some destructuring defaults support has been added, see [here](https://github.com/6to5/6to5/issues/230#issuecomment-68818054) for more info.

**Comment by sebmck** [comment]
Currently available as of 2.9.0
